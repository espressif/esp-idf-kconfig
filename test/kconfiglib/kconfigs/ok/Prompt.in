# Test prompts and conditions both explicit and inline
# explicit: prompt "prompt" if EXPRESSION
# inline: bool "prompt" if EXPRESSION
mainmenu "Main menu"

    config INLINE_SIMPLE
        bool "Inline simple prompt"
        default y

    config INLINE_CONDITIONED
        bool "Inline complex prompt" if INLINE_SIMPLE
        default y

    config INLINE_COMPLEX
        bool "Inline complex prompt" if INLINE_SIMPLE || SOMETHING_ELSE && !SOMETHING_MORE
        default y

    config EXPLICIT_SIMPLE
        bool
        prompt "Explicit simple prompt"
        default y

    config EXPLICIT_CONDITIONED
        bool
        prompt "Explicit complex prompt" if INLINE_SIMPLE
        default y

    config EXPLICIT_COMPLEX
        bool
        prompt "Explicit complex prompt" if !INLINE_SIMPLE && !SOMETHING_ELSE || SOMETHING_MORE
        default y

    config SPECIAL_CHARS
        bool "Special characters: _ {}"
        default y
