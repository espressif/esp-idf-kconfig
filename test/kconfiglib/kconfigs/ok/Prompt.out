#
# Automatically generated file. DO NOT EDIT.
# Espressif IoT Development Framework (ESP-IDF)  Project Configuration
#
# default:
CONFIG_INLINE_SIMPLE=y
# default:
CONFIG_INLINE_CONDITIONED=y
# default:
CONFIG_INLINE_COMPLEX=y
# default:
CONFIG_EXPLICIT_SIMPLE=y
# default:
CONFIG_EXPLICIT_CONDITIONED=y
# default:
CONFIG_EXPLICIT_COMPLEX=y
# default:
CONFIG_SPECIAL_CHARS=y
